src/main/java/org/xbill/DNS/DohResolver.java

```diff
@@ -366,10 +366,13 @@ public Duration getTimeout() {
         }
       }
     } catch (IOException ioe) {
-      try (InputStream es = conn.getErrorStream()) {
-        byte[] buf = new byte[4096];
-        while (es.read(buf) > 0) {
```

Comment:
It'd be enough/easier to read to write this as ` while (es != null && es.read(buf) > 0) {`. This works because try-with-resources handles `null` gracefully.